spec/unit/temporal/activity_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'temporal/activity'
+
+class TestSimpleActivity < Temporal::Activity; end
+
+class TestActivityWithCustomName < Temporal::Activity
+  activity_name 'custom-activity-name'
+end
+
+describe Temporal::Activity do
+  describe '._name' do
+    it 'returns class name by default' do
+      expect(TestSimpleActivity._name).to eq('TestSimpleActivity')
+    end
+
+    it 'returns class name when overriden' do
+      expect(TestActivityWithCustomName._name).to eq('custom-activity-name')
```

Comment:
I would write an integration test to validate the worker looks the activity up by this name.

Reply:
The main integration spec already does it, but I'll add another one (without a custom name) for clarity